pkgs/development/python-modules/nireports/default.nix

```
@@ -0,0 +1,97 @@
+{ lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+  hatch-vcs,
+  hatchling,
+  nipreps-versions,
+  importlib-resources,
+  matplotlib,
+  nibabel,
+  nilearn,
+  nipype,
+  numpy,
+  pandas,
+  pybids,
+  pyyaml,
+  seaborn,
+  svgutils,
+  templateflow,
+  pytestCheckHook,
+  pytest-xdist,
+}:
+
+buildPythonPackage rec {
+  pname = "nireports";
+  version = "23.2.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "nipreps";
+    repo = "nireports";
+    rev = version;
+    hash = "sha256-kr6nWsCA4GcPLEYV/IPm0N3VRz+Mho209/iNVd+59Ik=";
+  };
+
+  build-system = [
+    hatch-vcs
+    hatchling
+    nipreps-versions
+  ];
+
+  dependencies = [
+    matplotlib
+    nibabel
+    nilearn
+    nipype
+    numpy
+    pandas
+    pybids
+    pyyaml
+    seaborn
+    svgutils
+    templateflow
+  ] ++ lib.optionals (pythonOlder "3.12") [
+    importlib-resources
+  ];
+
+  env.SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeCheckInputs = [
+    matplotlib
+    pytestCheckHook
+    pytest-xdist
+  ];
+
+  pytestFlagsArray = [ "nireports" ];
```

```suggestion
  enabledTestPaths = [ "nireports" ];
```

BTW, the relevant attribute to specify extra pytest command-line arguments is `pytestFlags`.